gammapy/background/models.py

```diff
@@ -71,3 +80,820 @@ def evaluate(self, x, y):
+
+        data = table.as_array()
+
+        header = fits.Header()
+        header.update(table.meta)
+ 
+        tbhdu = fits.BinTableHDU(data, header, name=name)
+ 
+        # Copy over column meta-data
+        for colname in table.colnames:
+            tbhdu.columns[colname].unit = str(table[colname].unit)
+
+        # TODO: this method works fine but the order of keywords in the table
+        # header is not logical: for instnce, list of keywords with column
+        # units (TUNITi) is appended after the list of column keywords
+        # (TTYPEi, TFORMi), instead of in between.
+
+        return tbhdu
+
+
+    def to_fits_image(self):
+        """Convert cube background model to image fits format.
+
+        Returns
+        -------
+        hdu : `~astropy.io.fits.ImageHDU`
+            image containing the bg cube
+        """
+        hdu = fits.ImageHDU(data=self.background.value)
+        #TODO: store units (of bg) somewhere in header??!!!!
+        #TODO: implement WCS object to be able to read the det coords -> done
+        #TODO: energy binning: store in HDU table like for SpectralCube class
+
+        # get WCS object
+        wcs = self.det_WCS
+
+        # Now, write out the WCS object as a FITS header
+        wcs_header = wcs.to_header()
+
+        # transfering header values
+        # need to copy necessary values one by one
+        hdu.header['CTYPE1'] = wcs_header['CTYPE1']
+        hdu.header['CTYPE2'] = wcs_header['CTYPE2']
+        hdu.header['CUNIT1'] = wcs_header['CUNIT1']
+        hdu.header['CUNIT2'] = wcs_header['CUNIT2']
+        hdu.header['CDELT1'] = wcs_header['CDELT1']
+        hdu.header['CDELT2'] = wcs_header['CDELT2']
+        hdu.header['CRPIX1'] = wcs_header['CRPIX1']
+        hdu.header['CRPIX2'] = wcs_header['CRPIX2']
+        hdu.header['CRVAL1'] = wcs_header['CRVAL1']
+        hdu.header['CRVAL2'] = wcs_header['CRVAL2']
+
+        return hdu
+
+
+    def write_bin_table(self, *args, **kwargs):
+        """Write cube background model to binary table in fits file.
+
+        This function is expected to be called on a
+        `~astropy.io.fits.BinTableHDU` object.
+        It calls `~astropy.io.fits.BinTableHDU.writeto`,
+        forwarding all arguments.
+        """
+        self.to_fits_bin_table().writeto(*args, **kwargs)
+
+
+    def write_image(self, *args, **kwargs):
+        """Write cube background model to image in fits file.
+
+        This function is expected to be called on a
+        `~astropy.io.fits.ImageHDU` object.
+        It calls `~astropy.io.fits.ImageHDU.writeto`,
+        forwarding all arguments.
+        """
+        self.to_fits_image().writeto(*args, **kwargs)
+
+
+    @property
+    def image_extent(self):
+        """Image extent `(x_lo, x_hi, y_lo, y_hi)`.
+
+        Returns
+        -------
+        im_extent : `~astropy.coordinates.Angle`
+            array of bins with the image extent
+        """
+        bx = self.detx_bins.degree
```

Comment:
Ok, I'm using
```python
return Angle([bx[0, -1], by[0, -1]])
```
to simplify a bit. The `.degree` is needed, in case the units of x and y don't match.